File: app/src/main/java/com/botijasoftware/utils/animation/Keyframe.java
package com.botijasoftware.utils.animation;

import com.botijasoftware.utils.Vector3;

public class Keyframe {

	public float time;
	public Vector3 translate;
	public float angle;
	public Vector3 axis;
	public Vector3 scale;

	public Keyframe(float time) {
		this.time = time;
	}

	public void setTranslate(float x, float y, float z) {
		translate = new Vector3(x,y,z);

	}

	public void setScale(float x, float y, float z) {
		scale = new Vector3(x,y,z);

	}

	public void setRotation(float angle, float x, float y, float z) {
		this.angle = angle;
		axis = new Vector3(x, y, z);
	}

	public void transformBone(Bone bone) {
		bone.transmformMatrix(translate, axis, (float)(angle*Math.PI/180.0f), scale); 	
	}

}

File: app/src/main/java/com/botijasoftware/utils/animation/Keyframe.kt
package com.botijasoftware.utils.animation

import com.botijasoftware.utils.Vector3

class Keyframe(var time: Float) {
    lateinit var translate: Vector3
    var angle: Float = 0.toFloat()
    lateinit var axis: Vector3
    lateinit var scale: Vector3

    fun setTranslate(x: Float, y: Float, z: Float) {
        translate = Vector3(x, y, z)

    }

    fun setScale(x: Float, y: Float, z: Float) {
        scale = Vector3(x, y, z)

    }

    fun setRotation(angle: Float, x: Float, y: Float, z: Float) {
        this.angle = angle
        axis = Vector3(x, y, z)
    }

    fun transformBone(bone: Bone) {
        bone.transmformMatrix(translate, axis, (angle * Math.PI / 180.0f).toFloat(), scale)
    }

}
